mitmproxy/tools/web/app.py

```diff
@@ -290,13 +292,44 @@ def get(self):
 class WebSocketEventBroadcaster(tornado.websocket.WebSocketHandler):
     # raise an error if inherited class doesn't specify its own instance.
     connections: ClassVar[set[WebSocketEventBroadcaster]]
+    application: Application  # reference to the application, in order to be able to update the filters
 
     def open(self, *args, **kwargs):
         self.connections.add(self)
 
     def on_close(self):
         self.connections.discard(self)
 
+    async def on_message(self, message: str):
+        try:
+            data = json.loads(message)
+
+            # Validate the required fields
+            if not all(key in data for key in ["resource", "cmd", "name", "expr"]):
+                print(f"Invalid message format: Missing required fields.")
+                return
+
+            resource = data["resource"]
+            command = data["cmd"]
+            name = data["name"]
+            expression = data["expr"]
+
+            if resource == "flows" and command == "updateFilter":
+                filters_manager = self.application.master.filters_manager
+
+                filters_manager.update_filter(name, expression)
+                print(f"Updated filters: {filters_manager.get_all_filters()}")
+                self.broadcast(
+                    resource="flows",
+                    cmd="filtersUpdated",
+                    data=filters_manager.get_matching_flow_ids(),
```

Comment:
Just to be on the same page: `filtersUpdated` needs to contain the following:

1. The name of the filter that was just updated.
2. The current value of that filter.
3.  The list of flows that match that particular filter expression at the moment.

It does not need to send over the filter state of other existing filters.